### PR TITLE
Feature/justiceotuya/login

### PR DESCRIPTION
**Pull request by justiceotuya**
finished the login page

### CODE REVIEW

File: components/Authentication/components/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+ *  function that is used to display the Login Page
+ * @function
+ * @return {Object} the login page
+ */
+class NormalLoginForm extends React.Component {
+    state = {
+        iconLoading: false, // loading icon when code is accessing network
+        loading: false,
+    }
+
```
Comment (Way2nnadi) [inline components/Authentication/components/Login.jsx:31]: use the same es-doc pattern that you used to document the `NormalLoginForm` class to document your `enterLoading` and `handleSubmit` functions. Document all methods/function in react components expect `Lifecycle` methods.

**Review by Way2nnadi — DISMISSED** [review]
Looks great, need a bit of documentation, but overall great

File: components/Authentication/components/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                        </Form.Item>
+
+                        <Form.Item>
+                            <a className="login-form-forgot" href="/password_reset">Forgot password</a>
+                            <Button type="primary" htmlType="submit" className="login-form-button" loading={loading} onClick={this.enterLoading}>
+                                Log in
+                            </Button>
+                            Or
+                            <a className="login-form-register" href="/signup"> register now!</a>
+                        </Form.Item>
```
Comment (enyeInc) [inline components/Authentication/components/Login.jsx]: map over these `Form.Item`s

File: components/Authentication/components/Login.jsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+                                <Input prefix={<Icon type="lock" className="form_icon" />} type="password" placeholder="Password" />
+                            )}
+                        </Form.Item>
+
+                        <Form.Item>
+                            <a className="login-form-forgot" href="/password_reset">Forgot password</a>
+                            <Button type="primary" htmlType="submit" className="login-form-button" loading={loading} onClick={this.enterLoading}>
+                                Log in
+                            </Button>
+                            Or
```
Comment (enyeInc) [inline components/Authentication/components/Login.jsx:103]: Why is `Or` here?

Comment (justiceotuya) [reply components/Authentication/components/Login.jsx]: it is part of the UI, the statement is 'Log in or register now'

Comment (justiceotuya) [reply components/Authentication/components/Login.jsx:103]: > Why is `Or` here?

it is part of the UI, the statement is 'Log in or register now'

File: components/Authentication/components/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                        <Form.Item>
+                            {getFieldDecorator('password', {
+                                rules: [{
+                                    message: loginPasswordInputError,
+                                    required: true,
+                                }],
+                            })(
+                                <Input prefix={<Icon type="lock" className="form_icon" />} type="password" placeholder="Password" />
+                            )}
+                        </Form.Item>
```
Comment (enyeInc) [inline components/Authentication/components/Login.jsx:96]: map over these Form.Items. Example
```
const FORM_ITEMS = [{
    field: 'password',
    iconType: 'lock',
    iconClass: 'form_icon',
    inputType: 'password',
    placeholder: 'Password',
    rules: [{
        message: '...',
        required: true,
    }],
}, {...}]
...
FORM_ITEMS.map(item => {
    const { field, iconType, iconClass, inputType, placeholder, rules } = item;

    return (
        <Form.Item>
            {getFieldDecorator(field, { rules })(
                <Input
                    prefix={<Icon type={iconType} className={iconClass} />}
                    type={inputType}
                    placeholder={placeholder}
                />
            )}
        </Form.Item>
    )
});
```

**Pull request closed without merging**